src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final int figuresCount = 5;
+
+    public Figure getRandomFigure() {
+        int figureNumber = this.random.nextInt(this.figuresCount);
+        String color = this.colorSupplier.getRandomColor();
+
+        switch (figureNumber) {
+            case 0:
+                return new Square(color, this.random.nextDouble());
+            case 1:
+                return new Rectangle(color, this.random.nextDouble(), this.random.nextDouble());
+            case 2:
+                return new RightTriangle(color, this.random.nextDouble(), this.random.nextDouble());
+            case 3:
+                return new Circle(color, this.random.nextDouble());
+            case 4:
+                return new IsoscelesTrapezoid(
+                        color, this.random.nextDouble(),
+                        this.random.nextDouble(),
+                        this.random.nextDouble()
+                );
+            default:
+                return null;
```

Comment:
Returning null from a method is a bad practice. If you use a switch case construction in your solution, you may just put the last possible option in the default case.